From 9002df9bbf7f11006a6ffedeb003a4836d2a05ee Mon Sep 17 00:00:00 2001
From: Teresa Remmet <t.remmet@phytec.de>
Date: Mon, 27 Jul 2020 14:18:59 +0200
Subject: [PATCH] Revert "imx8m: Add DTB pre-process script to check u-boot
 DTB"

This reverts commit ab433440269bbd5383b47ee60957e9906cda0b3b.

This patch causes a lot of trouble for us it is only useful in NXP
context and breaks our u-boot image creation in multiple ways.

Signed-off-by: Teresa Remmet <t.remmet@phytec.de>
---
 iMX8M/soc.mak        | 29 +++++------------------------
 scripts/dtb_check.sh | 27 ---------------------------
 2 files changed, 5 insertions(+), 51 deletions(-)
 delete mode 100755 scripts/dtb_check.sh

diff --git a/iMX8M/soc.mak b/iMX8M/soc.mak
index b7b3986fb59d..5b79971ba0f3 100644
--- a/iMX8M/soc.mak
+++ b/iMX8M/soc.mak
@@ -19,7 +19,6 @@ ARCHIVE_NAME ?= $(shell cat nightly.txt).tar
 BITBUCKET_SERVER=https://bitbucket.sw.nxp.com
 DDR_FW_DIR=projects/IMX/repos/linux-firmware-imx/raw/firmware/ddr/synopsys
 PAD_IMAGE = ../scripts/pad_image.sh
-DTB_PREPROC = ../scripts/dtb_check.sh
 
 PRINT_FIT_HAB_OFFSET ?= 0x60000
 DEK_BLOB_LOAD_ADDR = 0x40400000
@@ -133,53 +132,38 @@ u-boot-atf-tee.bin: u-boot.bin bl31.bin tee.bin
 clean:
 	@rm -f $(MKIMG) u-boot-atf.bin u-boot-atf-tee.bin u-boot-spl-ddr.bin u-boot.itb u-boot.its u-boot-ddr3l.itb u-boot-ddr3l.its u-boot-spl-ddr3l.bin u-boot-ddr4.itb u-boot-ddr4.its u-boot-spl-ddr4.bin u-boot-ddr4-evk.itb u-boot-ivt.itb u-boot-ddr4-evk.its $(OUTIMG)
 
-dtbs = evk.dtb
-$(dtbs):
-	./$(DTB_PREPROC) $(PLAT)-evk.dtb $(dtbs)
-
+dtbs = fsl-$(PLAT)-evk.dtb
 u-boot.itb: $(dtbs)
 	./$(PAD_IMAGE) tee.bin
 	./$(PAD_IMAGE) bl31.bin
 	./$(PAD_IMAGE) u-boot-nodtb.bin $(dtbs)
 	DEK_BLOB_LOAD_ADDR=$(DEK_BLOB_LOAD_ADDR) TEE_LOAD_ADDR=$(TEE_LOAD_ADDR) ATF_LOAD_ADDR=$(ATF_LOAD_ADDR) ./mkimage_fit_atf.sh $(dtbs) > u-boot.its
 	./mkimage_uboot -E -p 0x3000 -f u-boot.its u-boot.itb
-	@rm -f u-boot.its $(dtbs)
-
-dtbs_ddr3l = valddr3l.dtb
-$(dtbs_ddr3l):
-	./$(DTB_PREPROC) $(PLAT)-ddr3l-$(VAL_BOARD).dtb $(dtbs_ddr3l)
+	@rm -f u-boot.its
 
+dtbs_ddr3l = fsl-$(PLAT)-ddr3l-$(VAL_BOARD).dtb
 u-boot-ddr3l.itb: $(dtbs_ddr3l)
 	./$(PAD_IMAGE) tee.bin
 	./$(PAD_IMAGE) bl31.bin
 	./$(PAD_IMAGE) u-boot-nodtb.bin $(dtbs_ddr3l)
 	DEK_BLOB_LOAD_ADDR=$(DEK_BLOB_LOAD_ADDR) TEE_LOAD_ADDR=$(TEE_LOAD_ADDR) ATF_LOAD_ADDR=$(ATF_LOAD_ADDR) ./mkimage_fit_atf.sh $(dtbs_ddr3l) > u-boot-ddr3l.its
 	./mkimage_uboot -E -p 0x3000 -f u-boot-ddr3l.its u-boot-ddr3l.itb
-	@rm -f u-boot.its $(dtbs_ddr3l)
-
-dtbs_ddr4 = valddr4.dtb
-$(dtbs_ddr4):
-	./$(DTB_PREPROC) $(PLAT)-ddr4-$(VAL_BOARD).dtb $(dtbs_ddr4)
 
+dtbs_ddr4 = fsl-$(PLAT)-ddr4-$(VAL_BOARD).dtb
 u-boot-ddr4.itb: $(dtbs_ddr4)
 	./$(PAD_IMAGE) tee.bin
 	./$(PAD_IMAGE) bl31.bin
 	./$(PAD_IMAGE) u-boot-nodtb.bin $(dtbs_ddr4)
 	DEK_BLOB_LOAD_ADDR=$(DEK_BLOB_LOAD_ADDR) TEE_LOAD_ADDR=$(TEE_LOAD_ADDR) ATF_LOAD_ADDR=$(ATF_LOAD_ADDR) ./mkimage_fit_atf.sh $(dtbs_ddr4) > u-boot-ddr4.its
 	./mkimage_uboot -E -p 0x3000 -f u-boot-ddr4.its u-boot-ddr4.itb
-	@rm -f u-boot.its $(dtbs_ddr4)
-
-dtbs_ddr4_evk = evkddr4.dtb
-$(dtbs_ddr4_evk):
-	./$(DTB_PREPROC) $(PLAT)-ddr4-evk.dtb $(dtbs_ddr4_evk)
 
+dtbs_ddr4_evk = fsl-$(PLAT)-ddr4-evk.dtb
 u-boot-ddr4-evk.itb: $(dtbs_ddr4_evk)
 	./$(PAD_IMAGE) tee.bin
 	./$(PAD_IMAGE) bl31.bin
 	./$(PAD_IMAGE) u-boot-nodtb.bin $(dtbs_ddr4_evk)
 	DEK_BLOB_LOAD_ADDR=$(DEK_BLOB_LOAD_ADDR) TEE_LOAD_ADDR=$(TEE_LOAD_ADDR) ATF_LOAD_ADDR=$(ATF_LOAD_ADDR) ./mkimage_fit_atf.sh $(dtbs_ddr4_evk) > u-boot-ddr4-evk.its
 	./mkimage_uboot -E -p 0x3000 -f u-boot-ddr4-evk.its u-boot-ddr4-evk.itb
-	@rm -f u-boot.its $(dtbs_ddr4_evk)
 
 ifeq ($(HDMI),yes)
 flash_evk: $(MKIMG) signed_hdmi_imx8m.bin u-boot-spl-ddr.bin u-boot.itb
@@ -256,21 +240,18 @@ print_fit_hab: u-boot-nodtb.bin bl31.bin $(dtbs)
 	./$(PAD_IMAGE) bl31.bin
 	./$(PAD_IMAGE) u-boot-nodtb.bin $(dtbs)
 	TEE_LOAD_ADDR=$(TEE_LOAD_ADDR) ATF_LOAD_ADDR=$(ATF_LOAD_ADDR) VERSION=$(VERSION) ./print_fit_hab.sh $(PRINT_FIT_HAB_OFFSET) $(dtbs)
-	@rm -f $(dtbs)
 
 print_fit_hab_ddr4: u-boot-nodtb.bin bl31.bin $(dtbs_ddr4_evk)
 	./$(PAD_IMAGE) tee.bin
 	./$(PAD_IMAGE) bl31.bin
 	./$(PAD_IMAGE) u-boot-nodtb.bin $(dtbs_ddr4_evk)
 	TEE_LOAD_ADDR=$(TEE_LOAD_ADDR) ATF_LOAD_ADDR=$(ATF_LOAD_ADDR) VERSION=$(VERSION) ./print_fit_hab.sh $(PRINT_FIT_HAB_OFFSET) $(dtbs_ddr4_evk)
-	@rm -f $(dtbs_ddr4_evk)
 
 print_fit_hab_flexspi: u-boot-nodtb.bin bl31.bin $(dtbs)
 	./$(PAD_IMAGE) tee.bin
 	./$(PAD_IMAGE) bl31.bin
 	./$(PAD_IMAGE) u-boot-nodtb.bin $(dtbs)
 	TEE_LOAD_ADDR=$(TEE_LOAD_ADDR) ATF_LOAD_ADDR=$(ATF_LOAD_ADDR) VERSION=$(VERSION) BOOT_DEV="flexspi" ./print_fit_hab.sh $(PRINT_FIT_HAB_OFFSET) $(dtbs)
-	@rm -f $(dtbs)
 
 nightly :
 	@echo "Pulling nightly for $(PLAT) evk board from $(SERVER)/$(DIR)"
diff --git a/scripts/dtb_check.sh b/scripts/dtb_check.sh
deleted file mode 100755
index ecf986d260aa..000000000000
--- a/scripts/dtb_check.sh
+++ /dev/null
@@ -1,27 +0,0 @@
-#!/bin/bash
-
-let dtba=0
-if [ -f $1 ]; then
-    let dtba=1
-fi
-
-if [ -f fsl-$1 ]; then
-    let dtba=$((dtba + 2))
-fi
-
-if [ $((dtba)) == 3 ]; then
-    echo " Two u-boot DTB files exist: fsl-"$1 "and" $1
-    echo " Please delete unused one!"
-    echo " u-boot imx_v2020.04: "$1
-    echo " u-boot imx_v2019.04: fsl-"$1
-    exit -1
-elif [ $((dtba)) == 0 ]; then
-    echo " Can't find u-boot DTB file, please copy from u-boot"
-    exit -2
-elif [ $((dtba)) == 1 ]; then
-    echo "Use u-boot DTB: "$1
-    cp -f $1 $2
-elif [ $((dtba)) == 2 ]; then
-    echo "Use u-boot DTB: fsl-"$1
-    cp -f fsl-$1 $2
-fi
-- 
2.7.4

